test/integ-new/main_new_test.go

```diff
@@ -212,47 +222,110 @@ func TestUIServiceHealthCheck(t *testing.T) {
 
-			ctx := context.Background()
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
 
-			uri := cfg.CertRotatorServiceAddr + tc.path
-			resp := testSendHTTPReq(ctx, t, uri, cfg.CertRotatorServiceIDToken)
+	kmsClient, err := kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to setup kms client: %s", err)
+	}
 
-			defer resp.Body.Close()
+	keyResouceName := fmt.Sprintf("projects/%s/locations/global/keyRings/%s/cryptoKeys/%s", cfg.ProjectID, cfg.KeyRing, cfg.KeyName)
 
-			if got, want := resp.StatusCode, tc.wantStatusCode; got != want {
-				b, err := io.ReadAll(resp.Body)
-				if err != nil {
-					t.Fatal(err)
-				}
-				t.Errorf("Got unexpected status code, got=%d want=%d, response=%s", got, want, string(b))
-			}
+	// Validate we have a single enabled key that is primary.
+	testValidateKeyVersionState(ctx, t, kmsClient, keyResouceName, 1,
+		map[int]kmspb.CryptoKeyVersion_CryptoKeyVersionState{
+			1: kmspb.CryptoKeyVersion_ENABLED,
 		})
-	}
+
+	t.Run("new_key_creation", func(t *testing.T) {
+		time.Sleep(5001 * time.Millisecond) // Wait past the next rotation event
+
+		testCallKeyRotationEndpoint(ctx, t)
+
+		// Validate we have created a new key, but haven't set it as primary yet.
+		testValidateKeyVersionState(ctx, t, kmsClient, keyResouceName, 1,
+			map[int]kmspb.CryptoKeyVersion_CryptoKeyVersionState{
+				1: kmspb.CryptoKeyVersion_ENABLED,
+				2: kmspb.CryptoKeyVersion_ENABLED,
+			})
+	})
+
+	t.Run("new_key_promotion", func(t *testing.T) {
+		time.Sleep(1001 * time.Millisecond) // Wait past the propagation delay.
+
+		testCallKeyRotationEndpoint(ctx, t)
+
+		// Validate our new key has been set to primary
+		testValidateKeyVersionState(ctx, t, kmsClient, keyResouceName, 2,
+			map[int]kmspb.CryptoKeyVersion_CryptoKeyVersionState{
+				1: kmspb.CryptoKeyVersion_ENABLED,
+				2: kmspb.CryptoKeyVersion_ENABLED,
+			})
+	})
+
+	t.Run("old_key_disable", func(t *testing.T) {
+		time.Sleep(2001 * time.Millisecond) // Wait past the grace period.
+
+		testCallKeyRotationEndpoint(ctx, t)
+
+		// Validate that our old key has been disabled.
+		testValidateKeyVersionState(ctx, t, kmsClient, keyResouceName, 2,
+			map[int]kmspb.CryptoKeyVersion_CryptoKeyVersionState{
+				1: kmspb.CryptoKeyVersion_DISABLED,
+				2: kmspb.CryptoKeyVersion_ENABLED,
+			})
+	})
+
+	t.Run("old_key_destroy", func(t *testing.T) {
+		time.Sleep(2001 * time.Millisecond) // Wait past the disabled period and next rotation event.
+
+		testCallKeyRotationEndpoint(ctx, t)
+
+		// Validate that our old key has been scheduled for destruction, and cycle has started again.
+		testValidateKeyVersionState(ctx, t, kmsClient, keyResouceName, 2,
+			map[int]kmspb.CryptoKeyVersion_CryptoKeyVersionState{
+				1: kmspb.CryptoKeyVersion_DESTROY_SCHEDULED,
+				2: kmspb.CryptoKeyVersion_ENABLED,
+				3: kmspb.CryptoKeyVersion_ENABLED,
+			})
+	})
+
+	t.Cleanup(func() {
+		testCleanUpKey(ctx, t, kmsClient, keyResouceName)
```

Comment:
If we cleanup here, would it affect api and public key tests?

Reply:
I have moved the kms key clean up back to the workflow, so it won't affect other tests.